indico/modules/events/registration/util.py

```diff
@@ -457,3 +440,37 @@ def generate_ticket_qr_code(registration):
     qr.add_data(json_qr_data)
     qr.make(fit=True)
     return qr.make_image()._img
+
+
+def get_open_or_registered_regforms(event, user):
+    """Get registration forms of an event which are open or user has already registered to.
+
+    :param event: the `Event` to get registrations for
+    :param user: A `User`
+    """
+    if not event.has_feature('registration'):
+        return []
+
+    return (RegistrationForm.query.with_parent(event)
+            .filter((RegistrationForm.is_open) |
+                    (RegistrationForm.registrations.any((Registration.user == user) & (~Registration.is_deleted))))
```

Comment:
The `()` around `~Registration.is_deleted` aren't needed - it's only around comparisons where the prededence of `&` and `|` would cause problems without them.